providers/darwin/machineid_darwin_amd64.go

```diff
@@ -0,0 +1,58 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build darwin,amd64,cgo
+
+package darwin
+
+// #include <unistd.h>
+// #include <uuid/uuid.h>
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+func MachineID() (string, error) {
```

Comment:
Can you add godocs for this that explain that it returns the hardware ID.

Reply:
done